static/src/javascripts/projects/common/utils/steady-page.js

```diff
@@ -0,0 +1,115 @@
+/*
+ Utility: steady-insert.js
+ Insert components into web-pages without jumping them around by
+ moving the current scroll position by the same distance as the height of the
+ inserted element.
+ Note: Inserted elements must have a fixed height when rendered
+ (i.e: no changing the height by loading an image, lazy-loading content etc)
+ */
+
+define([
+    'common/utils/fastdom-promise'
+], function (
+    fastdom
+) {
+    var currentFrameContainers = [];
+    var currentScrollPos;
+    var heightToAdd = 0;
+    var options = {
+        pixelGrace: 100 // Give the scroll some distance as it's likely the user has scrolled past the top of the screen
+    };
+
+    /*
+     * Height with margins
+     */
+    function outerHeight(el) {
+        var height = el.offsetHeight;
+        var style = getComputedStyle(el);
+
+        height += parseInt(style.marginTop) + parseInt(style.marginBottom);
+        return height;
+    }
+
+    /*
+     * The callback should contain the functionality
+     * to display the element (insertion, change of display).
+     */
+    function displayAndLayout (insertionCallback) {
+        return fastdom.write(function(){
+            insertionCallback();
+        });
+    }
+
+    /*
+     * Add the height of each inserted element within this animation frame
+     * together so we know what the total height change required for
+     * the scroll adjustment is
+     */
+    function getHeightAdjustment () {
+        return fastdom.read(function(){
```

Comment:
It needs to be read -> write -> read -> write because we can't read the heights until after we've inserted (and the initial read is to get the previous scroll position so it needs to be before the write)

\- *Read* the current window scroll position
-- *Write* the elements into the DOM
--- *Read* the heights
---- *Write* the new scroll position